### PR TITLE
Ban model update

### PR DESCRIPTION
**Pull request by TitanVJ**
walle pr: https://github.com/CSSS/wall_e/pull/439

### CODE REVIEW

File: wall_e_models/models.py [hunk 1]
```diff
@@ -45,35 +50,39 @@ def insert_records(cls, records: List[BanRecord]) -> None:
...
+            return False
+        return True
 
     @classmethod
     @sync_to_async
     def get_all_active_ban_user_ids(cls) -> List[int]:
         """Returns list of user_ids for all currently banned users"""
 
-        return list(BanRecord.objects.values_list('user_id', flat=True).filter(unban_date=None))
+        return list(BanRecord.objects.values_list('user_id', flat=True).filter(unban_date=0))
```
Comment (modernNeo) [inline wall_e_models/models.py:64]: im curious why this change? what difference does this make?

Comment (TitanVJ) [reply wall_e_models/models.py:64]: in postgres Null != Null so the default value is used now in the unban_date field. when unban_date is 0 its a currently active ban. if unban_date has a real date then obviously the user was unbanned.

Comment (TitanVJ) [reply wall_e_models/models.py:64]: unban_date is part of the unique constraint that prevents duplicate ban entries

Comment (modernNeo) [reply wall_e_models/models.py:64]: oh, I have realized what you are doing wrong.

all your filters should be written like this
`list(BanRecord.objects.filter(unban_date__isnull=True))`

Wish I had picked up on this before but I don't think I've ever even seen filter be done after `values_list` be used. and I would think it would defeat the whole idea you initially had of "using `values_list` to return only the necessary values so the query is faster" 🤔

**Review by modernNeo — CHANGES_REQUESTED** [review]
please redo your filtering syntax

File: wall_e_models/models.py (hunk 1 above)
Comment (TitanVJ) [reply wall_e_models/models.py:64]: no, its pointless to do that. the field unban_date used to be nullable now its not nullable. if the value of the field is null it cannot be part of the unique constraint, because null != null, which is required to prevent duplicate bans. so after the latest migration is migrated there won't be any unban_date values that are null for this reason.

Comment (modernNeo) [reply wall_e_models/models.py:64]: ah, I didn't notice that change
> the field unban_date used to be nullable now its not nullable. if the value of the field is null it cannot be part of the unique constraint

let me read through the code again

File: wall_e_models/models.py [hunk 2]
```diff
@@ -30,10 +31,14 @@ class BanRecord(models.Model):
...
     reason = models.CharField(max_length=512, null=False)
-    unban_date = models.BigIntegerField(null=True, default=None)
+    unban_date = models.BigIntegerField(null=False, default = 0)
 
     class Meta:
         db_table = 'wall_e_models_ban_records'
+        constraints = [
+            UniqueConstraint(fields=['user_id', 'unban_date'],
+                             name='unique_active_ban')
+        ]
```
Comment (modernNeo) [inline wall_e_models/models.py:41]:
```suggestion
        constraints = [
            UniqueConstraint(fields=['user_id'], name='unique_active_ban', condition=Q(unban_date__isnull=True))
        ]
```

doing this allow `unban_date` to remain `None` which is a better idea

File: wall_e_models/models.py (hunk 1 above)
Comment (modernNeo) [reply wall_e_models/models.py:64]: undo this change

Comment (modernNeo) [reply wall_e_models/models.py:64]: and all the other `filter` changes

File: wall_e_models/models.py [hunk 3]
```diff
@@ -45,35 +50,39 @@ def insert_records(cls, records: List[BanRecord]) -> None:
     @sync_to_async
     def insert_record(cls, record: BanRecord) -> None:
         """Adds entry to BanRecord table"""
-        record.save()
+        try:
+            record.save()
+        except Exception:
```
Comment (modernNeo) [inline wall_e_models/models.py:55]:
```suggestion
        except Exception:
        except IntegrityError:
```

broad Exception clauses are generally a bad ide

File: wall_e_models/migrations/0011_auto_20231021_0900.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('wall_e_models', '0010_embedavatar'),
+    ]
+
+    operations = [
```
Comment (modernNeo) [inline wall_e_models/migrations/0011_auto_20231021_0900.py:12]: please redo file with the suggested changes in `models.py`

**Review by modernNeo — CHANGES_REQUESTED** [review]

**Comment by modernNeo** [comment]
addressed with https://github.com/CSSS/wall_e_models/commit/2d0ee77a2c292e17b1da959785e7064361ab7536 and https://github.com/CSSS/wall_e_models/commit/8e3b5ea88184b8dce07185834ddc6a496967e66e
